Diagnostic/diagnostic.py

```diff
@@ -750,7 +754,20 @@ def install_omi():
 
     if need_install_omi:
         hutil.log("Begin omi installation.")
-        RunGetOutput(distConfig["installomi"])
+        isOmiInstalledSuccessfully = False
+        maxRetries = 3      # Retry up to 3 times to install OMI
+        for trialNum in range(1, maxRetries+1):
+            isOmiInstalledSuccessfully = RunGetOutput(distConfig["installomi"])[0] is 0
+            if isOmiInstalledSuccessfully:
+                break;
+            hutil.error("OMI install failed (trial #" + str(trialNum) + ").")
+            if trialNum < maxRetries:
```

Comment:
Needs to be <= maxRetries, otherwise you won't sleep when trialNum = maxRetries (the 3rd install, which is the 2nd retry).